service/controller/resource/pod/delete.go

```diff
@@ -41,6 +41,28 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		r.logger.Debugf(ctx, "found the current version of the reconciled pod in the Kubernetes API")
 	}
 
+	{
+		r.logger.LogCtx(ctx, "level", "debug", "message", "checking if cluster is being deleted")
+
+		clusterID, ok := currentPod.Labels[key.LabelCluster]
+		if !ok {
+			return microerror.Maskf(missingClusterLabelError, "pod is missing cluster label")
+		}
+
+		kvmConfig, err := r.g8sClient.ProviderV1alpha1().KVMConfigs(metav1.NamespaceDefault).Get(ctx, clusterID, metav1.GetOptions{})
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		if kvmConfig.DeletionTimestamp != nil {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "cluster is being deleted")
+			resourcecanceledcontext.SetCanceled(ctx)
```

Comment:
Is this necessary? I would have thought that just returning `nil` would do the trick here. There is no other action following here in this resource or am I missing something?

Reply:
We're [using](https://github.com/giantswarm/kvm-operator/blob/682a07b57f78d1742c27c2c42a858b7eefd77e51/service/controller/resource/pod/delete.go#L50-L52) this all across the file so I just kept things consistent. I don't know if it does anything though - we're calling `SetCanceled` in a lot of places in the operator but no one calls `IsCanceled`.